caddy.go

```diff
@@ -867,3 +867,19 @@ var (
 	// by default if no other file is specified.
 	DefaultConfigFile = "Caddyfile"
 )
+
+// contextKey is a value for use with context.WithValue. Inspired by the
+// net/http source.
+type contextKey struct {
+	name string
+}
```

Comment:
@ericdreeves FYI, I have merged the branch that contains httpserver.CtxKey so you can now use that.